src/mrvi/_model.py

```diff
@@ -743,13 +771,122 @@ def get_aggregated_posterior(
+            Only compute differential abundance for these sample labels.
+        batch_size
+            Minibatch size for computing the differential abundance.
+
+        Returns
+        -------
+        :class:`xarray.Dataset`
+            Returns an xarray.Dataset with data variables:
+            - `log_probs`: Array of shape (n_cells, n_samples) containing the log probabilities for each cell across samples.
+            - `{cov_key}_log_probs`: For each key in `sample_cov_keys`, an array of shape (n_cells, n_cov_values) containing the log probabilities for each cell across covariate values.
+        """
+        adata = self._validate_anndata(adata)
+        us = self.get_latent_representation(
+            adata, use_mean=True, give_z=False, batch_size=batch_size
+        )
+
+        log_probs = []
+        unique_samples = adata.obs[self.sample_key].unique()
+        for sample_name in tqdm(unique_samples):
+            ap = self.get_aggregated_posterior(
+                adata=adata, sample=sample_name, batch_size=batch_size
+            )
+            n_splits = max(adata.n_obs // batch_size, 1)
+            log_probs_ = []
+            for u_rep in np.array_split(us, n_splits):
+                log_probs_.append(
+                    jax.device_get(ap.log_prob(u_rep).sum(-1, keepdims=True))
+                )
+
+            log_probs.append(np.concatenate(log_probs_, axis=0))  # (n_cells, 1)
+
+        log_probs = np.concatenate(log_probs, 1)
+
+        coords = {
+            "cell_name": adata.obs_names.to_numpy(),
+            "sample": unique_samples,
+        }
+        data_vars = {
+            "log_probs": (["cell_name", "sample"], log_probs),
+        }
+        log_probs_arr = xr.Dataset(data_vars, coords=coords)
+
+        if sample_cov_keys is None or len(sample_cov_keys) == 0:
+            return log_probs_arr
+
+        sample_cov_log_probs_map = {}  # maps sample_cov_key to log_probs dataframe (n_cells, n_cov_values)
+        for sample_cov_key in sample_cov_keys:
+            sample_cov_unique_values = self.sample_info[sample_cov_key].unique()
+            per_val_log_probs = {}
+            for sample_cov_value in sample_cov_unique_values:
+                cov_samples = (
+                    self.sample_info[sample_cov_key] == sample_cov_value
+                ).index.to_numpy()
+                if sample_subset is not None:
+                    cov_samples = np.intersect1d(cov_samples, np.array(sample_subset))
+                if len(cov_samples) == 0:
+                    continue
+
+                sel_log_probs = log_probs_arr.log_probs.loc[{"sample": cov_samples}]
+                val_log_probs = logsumexp(sel_log_probs, axis=1) - np.log(
+                    sel_log_probs.shape[1]
+                )
+                per_val_log_probs[sample_cov_value] = val_log_probs
+            sample_cov_log_probs_map[sample_cov_key] = pd.DataFrame.from_dict(
+                per_val_log_probs
+            )
```

Comment:
We should catch cases where sample_cov_key is continuous

Reply:
An option could be to raise a ValueError whenever the covariate has more than some n_covs_limit value

Reply:
throwing a warning for this